### PR TITLE
Store avatar photos in s3

### PR DESCRIPTION
**Pull request by carlosascari**
For issue https://github.com/OpenCollective/OpenCollective/issues/39

Detects external avatars and attempts to upload them to Amazon S3 so as to have a reliable image source.

NOTE: the function added in lib, can be used for files other than images.

### CODE REVIEW

**Comment by xdamman** [comment]
Can you write tests for this?

**Comment by carlosascari** [comment]
@sedubois Is CircleCI configured to do Amazon S3 uploads? The last commit added a test that works locally, but gets 403 Forbidden during testing https://circleci.com/gh/OpenCollective/api/1512

**Comment by sedubois** [comment]
Would also be nice to squash the commits (either in advance or upon merge)

File: test/lib.imageUrlToAmazonUrl.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+var app = require('../index');
+var config = require('config');
+var imageUrlToAmazonUrl = require('../server/lib/imageUrlToAmazonUrl');
+var expect = require('chai').expect;
+
+var SAMPLE = 'https://d1ts43dypk8bqh.cloudfront.net/v1/avatars/1dca3d82-9c91-4d2a-8fc9-4a565c531764'
+
+describe('#Convert an external image url to a Amazon url', () => {
```
Comment (sedubois) [inline test/lib.imageUrlToAmazonUrl.js]: Please also prefix the describe with the filename as done in other tests

File: server/lib/imageUrlToAmazonUrl.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+// 
+// imageUrlToAmazonUrl(
+//	app.knox,
+// 	'http://avatars0.githubusercontent.com/u/13403593?v=3&s=200',
+// 	function(error, aws_src)
+// 	{
+// 		if (error) throw error
+// 		console.log('s3 src', aws_src);
+// 	}
+// )
```
Comment (sedubois) [inline server/lib/imageUrlToAmazonUrl.js]: can remove this unused code

**Comment by sedubois** [comment]
@carlosascari CircleCi has AWS keys in its env variables, so AWS in general should work. But I'm not familiar with that subject, how knox is set up etc. @xdamman any ideas?

Would also be nice to squash the commits (either in advance or upon merge)

**Comment by xdamman** [comment]
I'm not very familiar with knox but I'm sure Google can help :)

**Comment by carlosascari** [comment]
@sedubois I just noticed the line `Suppressing export of environment variables AWS_KEY, AWS_S3_BUCKET, AWS_SECRET, CLEARBIT_KEY, NODE_ENV, STRIPE_CLIENT_ID fork PR builds`, in the last failed test. 
IDK CircleCI, but that sounds like the cause of the 403, what do you think?

**Comment by sedubois** [comment]
@carlosascari I didn't notice that CircleCi message before, I wonder where it comes from. OTOH it says that it suppresses `NODE_ENV`, although it properly detects its value later on in `npm test`, so it seems it isn't really suppressing: `OpenCollective API listening at http://box954.localdomain:3060 in circleci environment.`. To be sure, maybe add some `console.log` of the vars in your tests?

**Comment by asood123** [comment]
Try it now

**Comment by carlosascari** [comment]
@asood123 CircleCI is still supressing the enviroment variables

**Pull request closed without merging**